src/main/java/graphql/schema/idl/ImmutableTypeDefinitionRegistry.java

```diff
@@ -49,6 +57,48 @@ public class ImmutableTypeDefinitionRegistry extends TypeDefinitionRegistry {
                 registry.schema,
                 registry.schemaParseOrder
         );
+        allImplementationsOf = calculateAllImplementsOf();
+        implementationsOf = calculateImplementationsOf(allImplementationsOf);
+    }
+
+    private Map<InterfaceTypeDefinition, List<ImplementingTypeDefinition>> calculateAllImplementsOf() {
+        ImmutableMap.Builder<InterfaceTypeDefinition, List<ImplementingTypeDefinition>> mapBuilder = ImmutableMap.builder();
+        List<ImplementingTypeDefinition> implementingTypeDefinitions = getTypes(ImplementingTypeDefinition.class);
+        for (TypeDefinition typeDef : types.values()) {
+            if (typeDef instanceof InterfaceTypeDefinition) {
+                InterfaceTypeDefinition interfaceTypeDef = (InterfaceTypeDefinition) typeDef;
+                ImmutableList.Builder<ImplementingTypeDefinition> listBuilder = ImmutableList.builder();
+                for (ImplementingTypeDefinition<?> implementingTypeDefinition : implementingTypeDefinitions) {
+                    List<Type> implementsList = implementingTypeDefinition.getImplements();
+                    for (Type iFace : implementsList) {
+                        Optional<InterfaceTypeDefinition> implementsAnInterface = getType(iFace, InterfaceTypeDefinition.class);
+                        if (implementsAnInterface.isPresent()) {
+                            boolean equals = implementsAnInterface.get().getName().equals(interfaceTypeDef.getName());
+                            if (equals) {
+                                listBuilder.add(implementingTypeDefinition);
+                                break;
+                            }
+                        }
+                    }
+                }
+                mapBuilder.put(interfaceTypeDef, listBuilder.build());
+            }
+        }
+        return mapBuilder.build();
+    }
```

Comment:
This now pre-computes the interface -> ImplementingTypeDefinition s at registry build time and hence later in the schema type checking stage, the repeated ask for interface information is much faster